vispy/visuals/colorbar.py

```diff
@@ -96,6 +96,105 @@
+
+    Note
+    ----
+    This is purely internal.
+    Externally, the ColorBarVisual must be used.
+    This class was separated out to encapsulate rendering information
+    That way, ColorBar simply becomes a CompoundVisual
+    """
+
+    def __init__(self, center_pos, halfdim,
+                 border_width=1.0,
+                 border_color="black",
+                 **kwargs):
+
+        self._center_pos = center_pos
+        self._halfdim = halfdim
+        self._border_width = border_width
+        self._border_color = border_color
+        self._border_color = Color(border_color)
+
+        Visual.__init__(self, vcode=VERT_SHADER_BORDER,
+                        fcode=FRAG_SHADER_BORDER, **kwargs)
+
+    @staticmethod
+    def _prepare_transforms(view):
+
+        program = view.shared_program
+        program.vert['visual_to_doc'] = \
+            view.transforms.get_transform('visual', 'document')
+        program.vert['doc_to_render'] = \
+            view.transforms.get_transform('document', 'render')
+
+    def _update(self):
+        x, y = self._center_pos
+        halfw, halfh = self._halfdim
+
+        border_vertices = np.array([
+            [x - halfw, y - halfh],
+            [x - halfw, y - halfh],
+
+            [x + halfw, y - halfh],
+            [x + halfw, y - halfh],
+
+            [x + halfw, y + halfh],
+            [x + halfw, y + halfh],
+
+            [x - halfw, y + halfh],
+            [x - halfw, y + halfh],
+
+            [x - halfw, y - halfh],
+            [x - halfw, y - halfh],
+        ], dtype=np.float32)
+
+        # Direction each vertex should move to correct for line width
+        adjust_dir = np.array([
+            [0, 0], [-1, -1],
+            [0, 0], [1, -1],
+            [0, 0], [1, 1],
+            [0, 0], [-1, 1],
+            [0, 0], [-1, -1],
+        ], dtype=np.float32)
+
+        self.shared_program['a_position'] = border_vertices
+        self.shared_program['a_adjust_dir'] = adjust_dir
+        self.shared_program.vert['border_width'] = self._border_width
+        self.shared_program.frag['border_color'] = self._border_color.rgba
+
+    def _prepare_draw(self, view=None):
+        self._update()
+        self._draw_mode = "triangle_strip"
+        return True
+
+    @property
+    def border_width(self):
+        """ The width of the border around the ColorBar in pixels
+        """
+        return self._border_width
+
+    @border_width.setter
+    def border_width(self, border_width):
+        self._border_width = border_width
+        # positions of text need to be changed accordingly
+        self._update()
+
+    @property
+    def border_color(self):
+        """ The color of the border around the ColorBar in pixels
+        """
+        return self._border_color
+
+    @border_color.setter
+    def border_color(self, border_color):
+        self._border_color = Color(border_color)
+        self.shared_program.frag['border_color'] = self._border_color.rgba
```

Comment:
Luke mentioned this too, that the `ViewBox` also renders borders. I'll take a look at it. Maybe in a separate PR?